### PR TITLE
Eth call fix3

### PR DESCRIPTION
**Pull request by lupin012**
This PR contains:
- fix on insufficient balance computation
- fix on default gas value if not present
- refactory evm_executor with add pre_check method
- aligns integration tests results

### CODE REVIEW

File: silkrpc/types/call.hpp [hunk 1]
```diff
@@ -42,7 +42,7 @@ struct Call {
         silkworm::Transaction txn{};
         txn.from = from;
         txn.to = to;
-        txn.gas_limit = gas.value_or(0);
+        txn.gas_limit = gas.value_or(30000000);
```
Comment (canepat) [inline silkrpc/types/call.hpp:45]: Add missing unit test for this as suggested by Codecov

File: silkrpc/core/evm_executor.cpp [hunk 2]
```diff
@@ -167,27 +192,15 @@ asio::awaitable<ExecutionResult> EVMExecutor::call(const silkworm::Block& block,
...
-                   asio::post(*context_.io_context, [exec_result, self = std::move(self)]() mutable {
-                      self.complete(exec_result);
-                   });
-                   return;
-                }
+                const intx::uint128 g0{silkworm::intrinsic_gas(txn, rev >= EVMC_HOMESTEAD, rev >= EVMC_ISTANBUL)};
+                assert(g0 <= UINT64_MAX); // true due to the precondition (transaction must be valid)
 
-                if (have < want) {
+                const auto error = pre_check(rev, txn, base_fee_per_gas, state, want, g0);
```
Comment (canepat) [inline silkrpc/core/evm_executor.cpp:200]: Passing `silkworm::EVM` as parameter to `pre_check` allows removing both `rev` and `state` that can be obtained from `evm` as follows:
```
const evmc_revision rev{evm.revision()};
const silkworm::IntraBlockState state{evm.state()};
```

**Review by canepat — CHANGES_REQUESTED** [review]
Empty line 224 in `evm_executor.cpp` should be removed
Line 230 in `evm_executor.cpp` has bad indentation

File: silkrpc/core/evm_executor.cpp (hunk 2 above)
Comment (lupin012) [reply silkrpc/core/evm_executor.cpp:200]: ok

File: silkrpc/core/evm_executor.cpp [hunk 3]
```diff
@@ -152,6 +152,31 @@ std::string EVMExecutor::get_error_message(int64_t error_code, const silkworm::B
     return error_message;
 }
 
+std::optional<std::string> pre_check(const evmc_revision rev, const silkworm::Transaction& txn, const intx::uint256 base_fee_per_gas, const silkworm::IntraBlockState & state,
```
[the remark this answers is not in the document]
Comment (lupin012) [reply silkrpc/core/evm_executor.cpp:155]: > ok

File: silkrpc/types/call.hpp (hunk 1 above)
Comment (lupin012) [reply silkrpc/types/call.hpp:45]: few tests added

**Review by canepat — APPROVED** [review]

**Pull request merged**